app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/PastOrderDetailsActivity.java

```diff
@@ -0,0 +1,35 @@
+package com.epfl.neighborfood.neighborfoodandroid.ui.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.epfl.neighborfood.neighborfoodandroid.R;
+import com.epfl.neighborfood.neighborfoodandroid.databinding.ActivityPastOrderDetailsBinding;
+import com.epfl.neighborfood.neighborfoodandroid.models.Order;
+
+public class PastOrderDetailsActivity extends AppCompatActivity{
+    Order order;
+    ActivityPastOrderDetailsBinding binding;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        Intent intent = getIntent();
+        order= (Order)intent.getSerializableExtra("order");
+        binding = ActivityPastOrderDetailsBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+        binding.mealImage.setImageResource(order.getMeal().getImageId());
+        binding.mealName.setText(order.getMeal().getName());
+        binding.mealDesc.setText(order.getMeal().getShortDescription());
+        binding.goVendorProfileId.setText(order.getMealVendor());
+        binding.goVendorProfileId.setOnClickListener(e->
+        {
+            Intent intentVendor = new Intent(PastOrderDetailsActivity.this, VendorProfileActivity.class);

```

Comment:
Why is there no extras (vendor id or name) linked to the intent? Don't we need to identify the corresponding vendor so we can display his profile?

Reply:
yes I'm planning to have that but for now since Vendor Profile is static I preferred to keep it this way and later on make the transition correctly to the VendorProfileActivity on a new task since it will take some time